lib/renderer/api/remote.js

```diff
@@ -201,106 +196,80 @@ const proxyFunctionProperties = function (remoteMemberFunction, metaId, name) {
     },
     getOwnPropertyDescriptor: (target, property) => {
       let descriptor = Object.getOwnPropertyDescriptor(target, property)
-      if (descriptor != null) return descriptor
+      if (descriptor) return descriptor
       loadRemoteProperties()
       return Object.getOwnPropertyDescriptor(target, property)
     }
   })
 }
 
 // Convert meta data from browser into real value.
-const metaToValue = function (meta) {
-  var el, i, len, ref1, results, ret
-  switch (meta.type) {
-    case 'value':
-      return meta.value
-    case 'array':
-      ref1 = meta.members
-      results = []
-      for (i = 0, len = ref1.length; i < len; i++) {
-        el = ref1[i]
-        results.push(metaToValue(el))
-      }
-      return results
-    case 'buffer':
-      return Buffer.from(meta.value)
-    case 'promise':
-      return resolvePromise({then: metaToValue(meta.then)})
-    case 'error':
-      return metaToPlainObject(meta)
-    case 'date':
-      return new Date(meta.value)
-    case 'exception':
-      throw new Error(meta.message + '\n' + meta.stack)
-    default:
-      if (remoteObjectCache.has(meta.id)) return remoteObjectCache.get(meta.id)
-
-      if (meta.type === 'function') {
-        // A shadow class to represent the remote function object.
-        let remoteFunction = function (...args) {
-          if (this && this.constructor === remoteFunction) {
-            // Constructor call.
-            let obj = ipcRenderer.sendSync('ELECTRON_BROWSER_CONSTRUCTOR', meta.id, wrapArgs(args))
-            // Returning object in constructor will replace constructed object
-            // with the returned object.
-            // http://stackoverflow.com/questions/1978049/what-values-can-a-constructor-return-to-avoid-returning-this
-            return metaToValue(obj)
-          } else {
-            // Function call.
-            let obj = ipcRenderer.sendSync('ELECTRON_BROWSER_FUNCTION_CALL', meta.id, wrapArgs(args))
-            return metaToValue(obj)
-          }
+function metaToValue (meta) {
+  const types = {
+    'value': () => meta.value,
+    'array': () => {
+      const members = meta.members
+      return members.map((member) => metaToValue(member))
+    },
+    'buffer': () => Buffer.from(meta.value),
+    'promise': () => resolvePromise({then: metaToValue(meta.then)}),
+    'error': () => metaToPlainObject(meta),
+    'date': () => new Date(meta.value),
+    'exception': () => new Error(`${meta.message}\n${meta.stack}`)
```

Comment:
old code throws the error, new code returns it. Does that cause any issues?

Reply:
i'm changing this back :)